docs/spec/enums.rst

```diff
@@ -0,0 +1,342 @@
+
+    def func2(any_color: Color):
+        reveal_type(any_color.value)  # Revealed type is Literal[1, 2, 3] (or int or object or Any)
+
+
+The value of ``_value_`` can be assigned in a constructor method. This technique
+is sometimes used to initialize both the member value and non-member attributes.
+If the value assigned in the class body is a tuple, the unpacked tuple value is
+passed to the constructor. Type checkers may validate consistency between assigned
+tuple values and the constructor signature::
+
+    class Planet(Enum):
+        def __init__(self, value: int, mass: float, radius: float):
+            self._value_ = value
+            self.mass = mass
+            self.radius = radius
+
+        MERCURY = (1, 3.303e+23, 2.4397e6)
+        VENUS = (2, 4.869e+24, 6.0518e6)
+        EARTH = (3, 5.976e+24, 6.37814e6)
+        MARS = (6.421e+23, 3.3972e6)  # Type checker error (optional)
+        JUPITER = 5  # Type checker error (optional)
+
+    reveal_type(Planet.MERCURY.value)  # Revealed type is Literal[1] (or int or object or Any)
+
+
+The class ``enum.auto`` and method ``_generate_next_value_`` can be used within
+an enum class to automatically generate values for enum members. Type checkers
+may support these to infer literal types for member values::
+
+    class Color(Enum):
+        RED = auto()
+        GREEN = auto()
+        BLUE = auto()
+
+    reveal_type(Color.RED.value)  # Revealed type is Literal[1] (or int or object or Any)
+
+
+If an enum class provides an explicit type annotation for ``_value_``, type
+checkers should enforce this declared type when values are assigned to
+``_value_``::
+
+    class Color(Enum):
+        _value_: int
+        RED = 1 # OK
+        GREEN = "green"  # Type error
+
+    class Planet(Enum):
+        _value_: str
+
+        def __init__(self, value: int, mass: float, radius: float):
+            self._value_ = value # Type error
+
+        MERCURY = (1, 3.303e+23, 2.4397e6)
+
+
+Enum Literal Expansion
+----------------------
+
+From the perspective of the type system, an enum class is equivalent to the union
+of the literal members within that enum. Because of this equivalency, the
+two types may be used interchangeably. Type checkers may therefore expand
+an enum type into a union of literal values during type narrowing and
+exhaustion detection::
+
+    class Color(Enum):
+        RED = 1
+        GREEN = 2
+        BLUE = 3
+    
+    def print_color1(c: Color):
+        if c is Color.RED or c is Color.BLUE:
+            print("red or blue")
+        else:
+            reveal_type(c)  # Revealed type is Literal[Color.GREEN]
+
+    def print_color2(c: Color):
+        match c:
+            case Color.RED | Color.BLUE:
+                print("red or blue")
+            case Color.GREEN:
+                print("green")
+            case _:
+                reveal_type(c)  # Revealed type is Never
+
+
+Likewise, a type checker should treat a complete union of all literal members
```

Comment:
Is "should" here a typo? In the first paragraph, type checkers "**may** therefore expand an enum type into a union," and then here you say "Likewise, a type checker **should**...". If "should" is correct here, the word "Likewise" seems wrong.